lib/pbench/server/api/resources/query_apis/dataset.py

```diff
@@ -0,0 +1,186 @@
+from typing import Any, Iterator
+
+from flask import current_app
+
+from pbench.server import JSONOBJECT, OperationCode, PbenchServerConfig
+from pbench.server.api.resources import (
+    ApiAuthorizationType,
+    ApiMethod,
+    ApiParams,
+    ApiSchema,
+    MissingParameters,
+    Parameter,
+    ParamType,
+    Schema,
+    UnauthorizedAdminAccess,
+)
+from pbench.server.api.resources.query_apis import ApiContext, ElasticBulkBase
+import pbench.server.auth.auth as Auth
+from pbench.server.cache_manager import CacheManager
+from pbench.server.database.models.audit import AuditType
+from pbench.server.database.models.datasets import (
+    Dataset,
+    OperationName,
+    OperationState,
+)
+from pbench.server.sync import Sync
+
+
+class Datasets(ElasticBulkBase):
```

Comment:
Yeah @webbnh. Initially, I thought the same, but a directory named `datasets` is present there. So If I call it `datasets`, then by default it will point to the file `datasets/__init__.py`.

Reply:
That's an interesting dilemma, @siddardh-ra.  I presume that you are right that we cannot name this module `datasets.py` if there is already a `datasets` directory.  However, given our existing directory structure (and API name space approach) and how Python maps modules to filenames, perhaps it would be appropriate to put this code in `datasets/__init__.py`.

That is, from a Python perspective, this likely is the fundamental class in the `pbench.server.api.resources.query_apis.datasets` package, so it would be entirely reasonable for its code to be found in the `__init__.py` module.  That is how single-module packages are arranged, and as additional modules added to the package, their files are added to the directory.  In our case, we're kind of backing into this, where have multiple modules and we are contracting the interface, but the result should be the same:  as we convert sibling APIs into separate methods on the same resource, it would make sense that they be moved into the central module.

Reply:
You're not entirely wrong, but "it's complicated". Both `IndexMapBase`, which lives in `pbench.server.api.resource.query_apis.datasets.__init__.py` and `ElasticBulkBase` depend on the index map, but they focus on different aspects. They *could* both extend a common index-map helper, and if `ElasticBulkBase` moved into the subdirectory then so would `query_apis.dataset.py` and `Datasets`.

That's a lot more refactoring than I'd like to see in this PR, however.